### PR TITLE
sys/shell: add loramac shell command

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR moves the `loramac` shell command implemented in the `tests/pkg_semtech-loramac` application to a more central place in `sys/shell`.

If one imports and sets the right variable/CFLAGS, it's possible to also get them with any application importing `shell_commands` module. For example, with `default`:

    USEPKG=semtech-loramac CFLAGS=-DREGION_EU868 BOARD=b-l072z-lrwan1 make -C examples/default

Note that `default`automatically imports the sx1276 module when using the B-L072Z-LRWAN1 board (via auto_init). For other application, one would have to also pass `USEMODULE=sx1276`.

### Issues/PRs references

~~This PR is based on #8639~~

~~Now based on #11583~~

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
#8639 got merged some time ago so I'm removing the dependency.

**Comment by fjmolinas** [comment]
@aabadie Couldn't you split this in to two PR's? For me the application refactor and to make it thread safe and adding the shell should be in different PR's.

**Comment by aabadie** [comment]
> Couldn't you split this in to two PR's? For me the application refactor and to make it thread safe and adding the shell should be in different PR's.

Well, it's just that this PR hasn't been rebased for a long time. It's done now.

File: sys/shell/commands/sc_loramac.c [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+        semtech_loramac_erase_config();
+    }
+#endif
+    else {
+        _loramac_usage();
+        return 1;
+    }
+
+    return 0;
+}
```
Comment (fjmolinas) [inline sys/shell/commands/sc_loramac.c]: No new line end of file

File: sys/auto_init/auto_init.c [hunk 2]
```diff
@@ -277,9 +277,14 @@ void auto_init(void)
 #endif
 
 #ifdef MODULE_SX127X
+#ifdef MODULE_SEMTECH_LORAMAC
```
Comment (fjmolinas) [inline sys/auto_init/auto_init.c]: IMO subordinate auto_init_loramac() to MODULE_SX127X is wrong, you are mixing up the mac layer and the network interface

**Comment by aabadie** [comment]
@fjmolinas regarding the application README, I prefer the separate PR: see #11426

File: sys/auto_init/auto_init.c (hunk 2 above)
Comment (fjmolinas) [reply sys/auto_init/auto_init.c]: I don't see why this should be in GRNC_NETIF, it has nothing to do with GNRC.

File: tests/pkg_semtech-loramac/Makefile [hunk 3]
```diff
@@ -14,6 +14,7 @@ LORA_REGION ?= EU868
 USEPKG += semtech-loramac
 USEMODULE += $(LORA_DRIVER)
 
+USEMODULE += auto_init_gnrc_netif
```
Comment (fjmolinas) [inline tests/pkg_semtech-loramac/Makefile]: I don't see why this should be in GRNC_NETIF, it has nothing to do with GNRC.

File: sys/auto_init/netif/auto_init_loramac.c [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+#ifdef MODULE_SEMTECH_LORAMAC
+
+#include "log.h"
+#include "semtech_loramac.h"
+
+semtech_loramac_t loramac;
+
+void auto_init_loramac(void)
```
Comment (fjmolinas) [inline sys/auto_init/netif/auto_init_loramac.c]: This should be a different PR, auto_init for LoRaMAC and the shell.

**Comment by fjmolinas** [comment]
Tested and its working except for the fact that `shell_commands` got removed from `examples_default`. But that should be addressed on a different PR. Do you want to wait for #11541 before merging this one?

**Comment by fjmolinas** [comment]
> nope, 

ACK then! :)

**Comment by aabadie** [comment]
It's all green @fjmolinas :)

**Comment by fjmolinas** [comment]
@aabadie Need to be rebased since #11541 was merged!

**Comment by aabadie** [comment]
Rebased (and still works hopefully)

**Comment by fjmolinas** [comment]
Still works, RE-ACK and GO!
